universal-application-tool-0.0.1/app/services/program/ProgramDefinition.java

```diff
@@ -2,12 +2,15 @@
 
 import com.google.auto.value.AutoValue;
 import com.google.common.collect.ImmutableList;
+import javax.annotation.Nullable;
+import models.Program;
 
 @AutoValue
 public abstract class ProgramDefinition {
 
   /** Unique identifier for a ProgramDefinition. */
-  public abstract long id();
+  @Nullable
```

Comment:
suggestion (blocking): Please use Optional if this can be null

Reply:
Done